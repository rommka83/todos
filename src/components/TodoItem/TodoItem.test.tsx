import React from 'react';
import renderer from 'react-test-renderer';
import { TodoItem } from './index';

describe('проверка TodoItem', () => {
  test('отрисовка выполненого TodoItem', () => {
    const component = renderer.create(<TodoItem value='выполнено' status />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('отрисовка невыполненого TodoItem', () => {
    const component = renderer.create(<TodoItem value='не выполнено' status={false} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
