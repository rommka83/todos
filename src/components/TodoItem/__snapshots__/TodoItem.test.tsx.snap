// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка TodoItem отрисовка выполненого TodoItem 1`] = `
<div
  className="root"
>
  <div
    className="status"
  >
    <svg
      color="var(--check)"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "var(--check)",
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 0 0-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z"
      />
    </svg>
  </div>
  <p
    className="root textCompleted"
  >
    выполнено
  </p>
</div>
`;

exports[`проверка TodoItem отрисовка невыполненого TodoItem 1`] = `
<div
  className="root"
>
  <div
    className="status"
  />
  <p
    className="root textMain"
  >
    не выполнено
  </p>
</div>
`;
